internal/config/sidecar.go

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package config providers configuration type and load configuration logic
+package config
+
+type Sidecar struct {
```

Comment:
![badge](https://img.shields.io/badge/review-imo-orange.svg?style=flat-square)
Currently, sidecar component is only for agent. however, in the future, other sidecar components may be required.
So it should be named as "AgentSidecar", "BlobStorageSidecar" or something others. "Sidecar" is too abstract here.

Reply:
I see, I prefer  AgentSidecar sounds good